### PR TITLE
dot-http: remove at 0.2.0

### PR DESCRIPTION
**Pull request by thenonameguy**
Fixed the build for ZHF (https://github.com/NixOS/nixpkgs/issues/199919) at #ZurichZHF :)

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by NickCao** [comment]
That's strange, doing HTTP requests to localhost should work even in the nix sandbox.

**Comment by thenonameguy** [comment]
@NickCao I might have misinterpreted the failing test:
https://hydra.nixos.org/build/199352692/nixlog/2

The app seemed to work locally though, so I am a bit baffled why the TCP connection failed

**Comment by NickCao** [comment]
Could be a bug in reqwest misinterpreting the meaning of localhost. Anyway this project seems effectively abandoned, what about dropping it in favor of https://github.com/Orange-OpenSource/hurl and friends? Also it blocks us from dropping openssl_1_1.

**Comment by mredaelli** [comment]
Agree to remove it

**Comment by thenonameguy** [comment]
I'm fine for deleting (as long there is 1 less broken package for 22.11 :tada: )
Should I do the removal in this PR?

**Comment by NickCao** [comment]
Fine, just rename the title.

**Review by mredaelli — APPROVED** [review]

**Comment by ajs124** [comment]
@thenonameguy can you add the entry to aliases.nix?

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -374,10 +374,11 @@ mapAliases ({
   docker-edge = throw "'docker-edge' has been removed, it was an alias for 'docker'"; # Added 2022-06-05
   dolphinEmu = dolphin-emu; # Added 2021-11-10
   dolphinEmuMaster = dolphin-emu-beta; # Added 2021-11-10
+  dot-http = throw "'dot-http' has been removed: abandoned by upstream. Use hurl instead."; # Added 2023-01-16
   dotnet-netcore = dotnet-runtime; # Added 2021-10-07
   double_conversion = throw "'double_conversion' has been renamed to/replaced by 'double-conversion'"; # Converted to throw 2022-02-22
-  dragon-drop = throw "'dragon-drop' has been removed in favor of 'xdragon'"; # Added 2022-04-10;
   draftsight = throw "draftsight has been removed, no longer available as freeware"; # Added 2020-08-14
+  dragon-drop = throw "'dragon-drop' has been removed in favor of 'xdragon'"; # Added 2022-04-10;
```
Comment (thenonameguy) [inline pkgs/top-level/aliases.nix:381]: This was in non-alphabetical order in this section. Figured it does not deserve it's own commit.
